src/code/z_map_disp.c

```diff
@@ -1,109 +1,1813 @@
+        if (dungeonMapRoomIter + 1 < D_801F56B0.rooms) {
+            D_801F56B0.unk84[dungeonMapRoomIter + 1] =
+                ALIGN16((intptr_t)D_801F56B0.unk84[dungeonMapRoomIter] + MapData_CPID_GetSizeOfMapTex(mapId));
+        } else {
+            heapNext = (intptr_t)D_801F56B0.unk84[dungeonMapRoomIter] + MapData_CPID_GetSizeOfMapTex(mapId);
+        }
+    }
+    for (roomIter = 0; roomIter < D_801BEC1C; roomIter++) {
+        entry = &D_801BEBB8.minimapList->entry[roomIter];
+        if (entry->mapId == 0xFFFF) {
+            D_801F56B0.unk104[roomIter] = NULL;
+        } else {
+            var_s3 = false;
+            mapCompactId = MapData_GetMapCompactId(entry->mapId);
+            for (dungeonMapRoomIter = 0; dungeonMapRoomIter < D_801F56B0.rooms; dungeonMapRoomIter++) {
+                if (mapCompactId == D_801F56B0.unk4[dungeonMapRoomIter]) {
+                    var_s3 = true;
+                    break;
+                }
+            }
+            if (!var_s3) {
+                D_801F56B0.unk104[roomIter] = NULL;
+            } else {
+                void* dummy = D_801F56B0.unk84[dungeonMapRoomIter]; //! FAKE:
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801064CC.s")
+                D_801F56B0.unk104[roomIter] = D_801F56B0.unk84[dungeonMapRoomIter];
+            }
+        }
+    }
+    return heapNext;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106530.s")
+s32 func_80106BEC(s32 arg0, f32 arg1) {
+    if (arg0 == 0) {
+        if ((D_801BEBB8.unk48[0] <= arg1) && ((D_801BEBB8.unk40 == 1) || (arg1 < D_801BEBB8.unk48[1]))) {
+            return 1;
+        }
+    } else if (arg0 >= (D_801BEBB8.unk40 - 1)) {
+        if (D_801BEBB8.unk48[D_801BEBB8.unk40 - 1] <= arg1) {
+            return 1;
+        }
+    } else if ((D_801BEBB8.unk48[arg0] <= arg1) && (arg1 < D_801BEBB8.unk48[arg0 + 1])) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_8010657C.s")
+s32 func_80106D08(s32 sceneId) {
+    switch (sceneId) {
+        case SCENE_MITURIN_BS:
+            return SCENE_MITURIN;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106644.s")
+        case SCENE_HAKUGIN_BS:
+            return SCENE_HAKUGIN;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_8010683C.s")
+        case SCENE_SEA_BS:
+            return SCENE_SEA;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801068B4.s")
+        case SCENE_INISIE_BS:
+            return SCENE_INISIE_N;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801068D8.s")
+        default:
+            return sceneId;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801068FC.s")
+#define COLOR16(r, g, b, a)                                                          \
+    (s16)(((s32)(((r)*31.0f) / 255.0f) << 11) | ((s32)(((g)*31.0f) / 255.0f) << 6) | \
+          ((s32)(((b)*31.0f) / 255.0f) << 1) | (a))
```

Comment:
I think this would be worth moving to `color.h`, although it could probably be called `RGBA16`. The name might need to indicate that floats are involved too?

Reply:
it isn't actually used for anything, it's more just part of an attempt to make `func_80106D5C` matching.